hnn_core/network.py

```diff
@@ -219,47 +277,182 @@ def __repr__(self):
                  len(self.pos_dict['L5_basket'])))
         return '<%s | %s>' % (class_name, s)
 
-    def _add_external_feeds(self):
-        """Legacy function, for backward compatibility with original HNN-GUI
+    def _create_drive_cells(self, target_populations, weights_by_receptor,
+                            location, space_constant, synaptic_delays,
+                            cell_specific=True):
+        if isinstance(synaptic_delays, dict):
+            for receptor in ['ampa', 'nmda']:
+                if not (
+                    set(list(weights_by_receptor[receptor].keys())).issubset(
+                        set(list(synaptic_delays[receptor].keys())))):
+                    raise ValueError(
+                        'synaptic_delays is either common (float), or needs '
+                        'to be specified for each non-zero AMPA/NMDA weight')
+
+        next_gid = self._n_gids
+        drive_gids = []
+        drive_cells = []
+        if cell_specific:
+            for cellname in target_populations:
```

Comment:
It seems to me you can avoid this huge `if`-`else` block by carefully setting what you iterate over. But I didn't look too closely, so I may be wrong